src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -218,6 +223,10 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
                         throw new CommandLineException(
                             LocalizedResourceManager.GetString(nameof(NuGetResources.Error_MsBuildTimedOut)));
                     }
+                    if (verbosity == Verbosity.Detailed)
+                    {
+                        await outputTask;
```

Comment:
Why await only if verbosity is detailed? There could be other warnings that need to show up consistently, right?